### ISSUE TITLE
kubeadm based kubernetes (v1.24.2) worker nodes are still in "NotReady" status even after installing calico CNI ("Error querying BIRD")

### ISSUE THREAD

**Issue by allankamau** [projectcalico/calico#6626]
kubeadm based kubernetes (v1.24.2) worker nodes are still in "NotReady" status even after installing calico CNI ("Error querying BIRD")

kubeadm based kubernetes (v1.24.2) worker nodes are still in "NotReady" status even after installing calico CNI ("dial unix /var/run/bird/bird.ctl: connect: no such file or directory")

I have deployed calico CNI on kubeadm based kubernetes cluster as per the document "https://projectcalico.docs.tigera.io/getting-started/kubernetes/self-managed-onprem/onpremises" but the worker nodes still have the "NotReady" status value.

TCP/IP port 179 is open on all the nodes.
The SELinux reports no denials.

The control plane node has status value of "Ready" while the worker nodes have the status value of "NotReady".

The contents of "/etc/cni/net.d/10-calico.conflist" on the controlplane and worker nodes are identical.

```
$ cat /etc/cni/net.d/10-calico.conflist

{
  "name": "k8s-pod-network",
  "cniVersion": "0.3.1",
  "plugins": [
    {
      "type": "calico",
      "datastore_type": "kubernetes",
      "mtu": 0,
      "nodename_file_optional": false,
      "log_level": "Info",
      "log_file_path": "/var/log/calico/cni/cni.log",
      "ipam": { "type": "calico-ipam", "assign_ipv4" : "true", "assign_ipv6" : "false"},
      "container_settings": {
          "allow_ip_forwarding": false
      },
      "policy": {
          "type": "k8s"
      },
      "kubernetes": {
          "k8s_api_root":"https://172.21.0.1:443",
          "kubeconfig": "/etc/cni/net.d/calico-kubeconfig"
      }
    },
    {
      "type": "bandwidth",
      "capabilities": {"bandwidth": true}
    },
    {"type": "portmap", "snat": true, "capabilities": {"portMappings": true}}
  ]
}
```

The kubelet service in one of the worker node has the following log output snipped

```
$ journalctl -x -u kubelet.service -f;

Aug 26 16:51:35 centos7-03-08 kubelet[2039]: I0826 16:51:35.419281    2039 kuberuntime_manager.go:1095] "Updating runtime config through cri with podcidr" CIDR="172.22.1.0/24"
Aug 26 16:51:35 centos7-03-08 kubelet[2039]: I0826 16:51:35.422753    2039 kubelet_network.go:60] "Updating Pod CIDR" originalPodCIDR="" newPodCIDR="172.22.1.0/24"
Aug 26 16:51:35 centos7-03-08 kubelet[2039]: E0826 16:51:35.423723    2039 kubelet.go:2349] "Container runtime network not ready" networkReady="NetworkReady=false reason:NetworkPluginNotReady message:Network plugin returns error: No CNI configuration file in /etc/cni/net.d/. Has your network provider started?"
Aug 26 16:51:35 centos7-03-08 kubelet[2039]: I0826 16:51:35.469909    2039 apiserver.go:52] "Watching apiserver"
Aug 26 16:51:35 centos7-03-08 kubelet[2039]: I0826 16:51:35.474825    2039 kubelet.go:2072] "SyncLoop ADD" source="api" pods=[kube-system/kube-proxy-z4ffn]
Aug 26 16:51:35 centos7-03-08 kubelet[2039]: I0826 16:51:35.474876    2039 topology_manager.go:200] "Topology Admit Handler"
Aug 26 16:51:35 centos7-03-08 kubelet[2039]: I0826 16:51:35.519672    2039 reconciler.go:270] "operationExecutor.VerifyControllerAttachedVolume started for volume \"kube-proxy\" (UniqueName: \"kubernetes.io/configmap/1177c68d-2206-438a-a8a4-4650968f6fcc-kube-proxy\") pod \"kube-proxy-z4ffn\" (UID: \"1177c68d-22
Aug 26 16:51:35 centos7-03-08 kubelet[2039]: I0826 16:51:35.519751    2039 reconciler.go:270] "operationExecutor.VerifyControllerAttachedVolume started for volume \"xtables-lock\" (UniqueName: \"kubernetes.io/host-path/1177c68d-2206-438a-a8a4-4650968f6fcc-xtables-lock\") pod \"kube-proxy-z4ffn\" (UID: \"1177c68
Aug 26 16:51:35 centos7-03-08 kubelet[2039]: I0826 16:51:35.519790    2039 reconciler.go:270] "operationExecutor.VerifyControllerAttachedVolume started for volume \"lib-modules\" (UniqueName: \"kubernetes.io/host-path/1177c68d-2206-438a-a8a4-4650968f6fcc-lib-modules\") pod \"kube-proxy-z4ffn\" (UID: \"1177c68d-
Aug 26 16:51:35 centos7-03-08 kubelet[2039]: I0826 16:51:35.519826    2039 reconciler.go:270] "operationExecutor.VerifyControllerAttachedVolume started for volume \"kube-api-access-cmdkq\" (UniqueName: \"kubernetes.io/projected/1177c68d-2206-438a-a8a4-4650968f6fcc-kube-api-access-cmdkq\") pod \"kube-proxy-z4ffn
Aug 26 16:51:35 centos7-03-08 kubelet[2039]: I0826 16:51:35.519841    2039 reconciler.go:157] "Reconciler: start to sync state"
Aug 26 16:51:35 centos7-03-08 kubelet[2039]: I0826 16:51:35.620137    2039 reconciler.go:245] "operationExecutor.MountVolume started for volume \"kube-proxy\" (UniqueName: \"kubernetes.io/configmap/1177c68d-2206-438a-a8a4-4650968f6fcc-kube-proxy\") pod \"kube-proxy-z4ffn\" (UID: \"1177c68d-2206-438a-a8a4-465096
Aug 26 16:51:35 centos7-03-08 kubelet[2039]: I0826 16:51:35.620230    2039 reconciler.go:245] "operationExecutor.MountVolume started for volume \"xtables-lock\" (UniqueName: \"kubernetes.io/host-path/1177c68d-2206-438a-a8a4-4650968f6fcc-xtables-lock\") pod \"kube-proxy-z4ffn\" (UID: \"1177c68d-2206-438a-a8a4-46
Aug 26 16:51:35 centos7-03-08 kubelet[2039]: I0826 16:51:35.620277    2039 reconciler.go:245] "operationExecutor.MountVolume started for volume \"lib-modules\" (UniqueName: \"kubernetes.io/host-path/1177c68d-2206-438a-a8a4-4650968f6fcc-lib-modules\") pod \"kube-proxy-z4ffn\" (UID: \"1177c68d-2206-438a-a8a4-4650
Aug 26 16:51:35 centos7-03-08 kubelet[2039]: I0826 16:51:35.620328    2039 reconciler.go:245] "operationExecutor.MountVolume started for volume \"kube-api-access-cmdkq\" (UniqueName: \"kubernetes.io/projected/1177c68d-2206-438a-a8a4-4650968f6fcc-kube-api-access-cmdkq\") pod \"kube-proxy-z4ffn\" (UID: \"1177c68d
Aug 26 16:51:35 centos7-03-08 kubelet[2039]: I0826 16:51:35.621807    2039 operation_generator.go:703] "MountVolume.SetUp succeeded for volume \"xtables-lock\" (UniqueName: \"kubernetes.io/host-path/1177c68d-2206-438a-a8a4-4650968f6fcc-xtables-lock\") pod \"kube-proxy-z4ffn\" (UID: \"1177c68d-2206-438a-a8a4-465
Aug 26 16:51:35 centos7-03-08 kubelet[2039]: I0826 16:51:35.621939    2039 operation_generator.go:703] "MountVolume.SetUp succeeded for volume \"lib-modules\" (UniqueName: \"kubernetes.io/host-path/1177c68d-2206-438a-a8a4-4650968f6fcc-lib-modules\") pod \"kube-proxy-z4ffn\" (UID: \"1177c68d-2206-438a-a8a4-46509
Aug 26 16:51:35 centos7-03-08 kubelet[2039]: I0826 16:51:35.622858    2039 operation_generator.go:703] "MountVolume.SetUp succeeded for volume \"kube-proxy\" (UniqueName: \"kubernetes.io/configmap/1177c68d-2206-438a-a8a4-4650968f6fcc-kube-proxy\") pod \"kube-proxy-z4ffn\" (UID: \"1177c68d-2206-438a-a8a4-4650968
Aug 26 16:51:35 centos7-03-08 kubelet[2039]: I0826 16:51:35.640167    2039 operation_generator.go:703] "MountVolume.SetUp succeeded for volume \"kube-api-access-cmdkq\" (UniqueName: \"kubernetes.io/projected/1177c68d-2206-438a-a8a4-4650968f6fcc-kube-api-access-cmdkq\") pod \"kube-proxy-z4ffn\" (UID: \"1177c68d-
Aug 26 16:51:35 centos7-03-08 kubelet[2039]: I0826 16:51:35.806429    2039 kuberuntime_manager.go:469] "No sandbox for pod can be found. Need to start a new one" pod="kube-system/kube-proxy-z4ffn"
Aug 26 16:51:36 centos7-03-08 kubelet[2039]: E0826 16:51:36.727504    2039 kubelet.go:2349] "Container runtime network not ready" networkReady="NetworkReady=false reason:NetworkPluginNotReady message:Network plugin returns error: No CNI configuration file in /etc/cni/net.d/. Has your network provider started?"
Aug 26 16:51:40 centos7-03-08 kubelet[2039]: W0826 16:51:40.287413    2039 manager.go:1176] Failed to process watch event {EventType:0 Name:/kubepods.slice/kubepods-besteffort.slice/kubepods-besteffort-pod1177c68d_2206_438a_a8a4_4650968f6fcc.slice/crio-87ae94f0d7f3fdee4e0a1ae10ad019be559d28c6387eb13c2f5370b744e
Aug 26 16:51:40 centos7-03-08 kubelet[2039]: I0826 16:51:40.298085    2039 provider.go:102] Refreshing cache for provider: *credentialprovider.defaultDockerConfigProvider
Aug 26 16:51:40 centos7-03-08 kubelet[2039]: I0826 16:51:40.298256    2039 provider.go:82] Docker config file not found: couldn't find valid .dockercfg after checking in [/var/lib/kubelet   /]
Aug 26 16:51:40 centos7-03-08 kubelet[2039]: I0826 16:51:40.738667    2039 kubelet.go:2110] "SyncLoop (PLEG): event for pod" pod="kube-system/kube-proxy-z4ffn" event=&{ID:1177c68d-2206-438a-a8a4-4650968f6fcc Type:ContainerStarted Data:87ae94f0d7f3fdee4e0a1ae10ad019be559d28c6387eb13c2f5370b744ef5792}
Aug 26 16:51:41 centos7-03-08 kubelet[2039]: E0826 16:51:41.728478    2039 kubelet.go:2349] "Container runtime network not ready" networkReady="NetworkReady=false reason:NetworkPluginNotReady message:Network plugin returns error: No CNI configuration file in /etc/cni/net.d/. Has your network provider started?"
Aug 26 16:51:46 centos7-03-08 kubelet[2039]: E0826 16:51:46.729843    2039 kubelet.go:2349] "Container runtime network not ready" networkReady="NetworkReady=false reason:NetworkPluginNotReady message:Network plugin returns error: No CNI configuration file in /etc/cni/net.d/. Has your network provider started?"
Aug 26 16:51:50 centos7-03-08 kubelet[2039]: I0826 16:51:50.756072    2039 kubelet.go:2110] "SyncLoop (PLEG): event for pod" pod="kube-system/kube-proxy-z4ffn" event=&{ID:1177c68d-2206-438a-a8a4-4650968f6fcc Type:ContainerStarted Data:7100a65cfd8071fa62d0a14774bb9f3650519e33204ad3c697cde479016611a7}
Aug 26 16:51:51 centos7-03-08 kubelet[2039]: E0826 16:51:51.731196    2039 kubelet.go:2349] "Container runtime network not ready" networkReady="NetworkReady=false reason:NetworkPluginNotReady message:Network plugin returns error: No CNI configuration file in /etc/cni/net.d/. Has your network provider started?"


Aug 26 17:06:42 centos7-03-08 kubelet[2039]: W0826 17:06:42.144069    2039 driver-call.go:149] FlexVolume: driver call failed: executable: /usr/libexec/kubernetes/kubelet-plugins/volume/exec/nodeagent~uds/uds, args: [init], error: fork/exec /usr/libexec/kubernetes/kubelet-plugins/volume/exec/nodeagent~uds/uds: 
Aug 26 17:06:42 centos7-03-08 kubelet[2039]: E0826 17:06:42.144091    2039 plugins.go:740] "Error dynamically probing plugins" err="error creating Flexvolume plugin from directory nodeagent~uds, skipping. Error: unexpected end of JSON input"
Aug 26 17:06:42 centos7-03-08 kubelet[2039]: E0826 17:06:42.144440    2039 driver-call.go:262] Failed to unmarshal output for command: init, output: "", error: unexpected end of JSON input
Aug 26 17:06:42 centos7-03-08 kubelet[2039]: W0826 17:06:42.144467    2039 driver-call.go:149] FlexVolume: driver call failed: executable: /usr/libexec/kubernetes/kubelet-plugins/volume/exec/nodeagent~uds/uds, args: [init], error: fork/exec /usr/libexec/kubernetes/kubelet-plugins/volume/exec/nodeagent~uds/uds: 
Aug 26 17:06:42 centos7-03-08 kubelet[2039]: E0826 17:06:42.144490    2039 plugins.go:740] "Error dynamically probing plugins" err="error creating Flexvolume plugin from directory nodeagent~uds, skipping. Error: unexpected end of JSON input"
Aug 26 17:06:42 centos7-03-08 kubelet[2039]: E0826 17:06:42.144835    2039 driver-call.go:262] Failed to unmarshal output for command: init, output: "", error: unexpected end of JSON input
Aug 26 17:06:42 centos7-03-08 kubelet[2039]: W0826 17:06:42.144865    2039 driver-call.go:149] FlexVolume: driver call failed: executable: /usr/libexec/kubernetes/kubelet-plugins/volume/exec/nodeagent~uds/uds, args: [init], error: fork/exec /usr/libexec/kubernetes/kubelet-plugins/volume/exec/nodeagent~uds/uds: 
Aug 26 17:06:42 centos7-03-08 kubelet[2039]: E0826 17:06:42.144896    2039 plugins.go:740] "Error dynamically probing plugins" err="error creating Flexvolume plugin from directory nodeagent~uds, skipping. Error: unexpected end of JSON input"
Aug 26 17:06:42 centos7-03-08 kubelet[2039]: I0826 17:06:42.152011    2039 operation_generator.go:703] "MountVolume.SetUp succeeded for volume \"node-certs\" (UniqueName: \"kubernetes.io/secret/3c950a6b-b5b3-4750-904b-ba51034aa65b-node-certs\") pod \"calico-node-jmlbp\" (UID: \"3c950a6b-b5b3-4750-904b-ba51034aa
Aug 26 17:06:42 centos7-03-08 kubelet[2039]: E0826 17:06:42.152820    2039 driver-call.go:262] Failed to unmarshal output for command: init, output: "", error: unexpected end of JSON input
Aug 26 17:06:42 centos7-03-08 kubelet[2039]: W0826 17:06:42.152840    2039 driver-call.go:149] FlexVolume: driver call failed: executable: /usr/libexec/kubernetes/kubelet-plugins/volume/exec/nodeagent~uds/uds, args: [init], error: fork/exec /usr/libexec/kubernetes/kubelet-plugins/volume/exec/nodeagent~uds/uds: 
Aug 26 17:06:42 centos7-03-08 kubelet[2039]: E0826 17:06:42.152866    2039 plugins.go:740] "Error dynamically probing plugins" err="error creating Flexvolume plugin from directory nodeagent~uds, skipping. Error: unexpected end of JSON input"
Aug 26 17:06:42 centos7-03-08 kubelet[2039]: I0826 17:06:42.359850    2039 kuberuntime_manager.go:469] "No sandbox for pod can be found. Need to start a new one" pod="calico-system/calico-node-jmlbp"
Aug 26 17:06:43 centos7-03-08 kubelet[2039]: I0826 17:06:43.454262    2039 kubelet.go:2110] "SyncLoop (PLEG): event for pod" pod="calico-system/calico-node-jmlbp" event=&{ID:3c950a6b-b5b3-4750-904b-ba51034aa65b Type:ContainerStarted Data:497048b7f0244f983e62db21b01b956d97c50ed5ed652f5e61ed0a3feb8880e5}
Aug 26 17:06:46 centos7-03-08 kubelet[2039]: E0826 17:06:46.983743    2039 kubelet.go:2349] "Container runtime network not ready" networkReady="NetworkReady=false reason:NetworkPluginNotReady message:Network plugin returns error: No CNI configuration file in /etc/cni/net.d/. Has your network provider started?"
Aug 26 17:06:51 centos7-03-08 kubelet[2039]: E0826 17:06:51.985207    2039 kubelet.go:2349] "Container runtime network not ready" networkReady="NetworkReady=false reason:NetworkPluginNotReady message:Network plugin returns error: No CNI configuration file in /etc/cni/net.d/. Has your network provider started?"
Aug 26 17:06:56 centos7-03-08 kubelet[2039]: E0826 17:06:56.986506    2039 kubelet.go:2349] "Container runtime network not ready" networkReady="NetworkReady=false reason:NetworkPluginNotReady message:Network plugin returns error: No CNI configuration file in /etc/cni/net.d/. Has your network provider started?"
Aug 26 17:07:01 centos7-03-08 kubelet[2039]: I0826 17:07:01.527605    2039 generic.go:296] "Generic (PLEG): container finished" podID=3c950a6b-b5b3-4750-904b-ba51034aa65b containerID="e1d98ac56154242d20eb12ad6ac25655ab5be547f155c0b8d37f7dbb72e246d4" exitCode=0
Aug 26 17:07:01 centos7-03-08 kubelet[2039]: I0826 17:07:01.527667    2039 kubelet.go:2110] "SyncLoop (PLEG): event for pod" pod="calico-system/calico-node-jmlbp" event=&{ID:3c950a6b-b5b3-4750-904b-ba51034aa65b Type:ContainerDied Data:e1d98ac56154242d20eb12ad6ac25655ab5be547f155c0b8d37f7dbb72e246d4}
Aug 26 17:07:01 centos7-03-08 kubelet[2039]: I0826 17:07:01.552176    2039 plugins.go:650] "Loaded volume plugin" pluginName="nodeagent/uds"
Aug 26 17:07:01 centos7-03-08 kubelet[2039]: E0826 17:07:01.987861    2039 kubelet.go:2349] "Container runtime network not ready" networkReady="NetworkReady=false reason:NetworkPluginNotReady message:Network plugin returns error: No CNI configuration file in /etc/cni/net.d/. Has your network provider started?"
Aug 26 17:07:06 centos7-03-08 kubelet[2039]: E0826 17:07:06.989177    2039 kubelet.go:2349] "Container runtime network not ready" networkReady="NetworkReady=false reason:NetworkPluginNotReady message:Network plugin returns error: No CNI configuration file in /etc/cni/net.d/. Has your network provider started?"
Aug 26 17:07:11 centos7-03-08 kubelet[2039]: E0826 17:07:11.991195    2039 kubelet.go:2349] "Container runtime network not ready" networkReady="NetworkReady=false reason:NetworkPluginNotReady message:Network plugin returns error: No CNI configuration file in /etc/cni/net.d/. Has your network provider started?"
Aug 26 17:07:16 centos7-03-08 kubelet[2039]: E0826 17:07:16.992265    2039 kubelet.go:2349] "Container runtime network not ready" networkReady="NetworkReady=false reason:NetworkPluginNotReady message:Network plugin returns error: No CNI configuration file in /etc/cni/net.d/. Has your network provider started?"
Aug 26 17:07:21 centos7-03-08 kubelet[2039]: E0826 17:07:21.994253    2039 kubelet.go:2349] "Container runtime network not ready" networkReady="NetworkReady=false reason:NetworkPluginNotReady message:Network plugin returns error: No CNI configuration file in /etc/cni/net.d/. Has your network provider started?"
Aug 26 17:07:26 centos7-03-08 kubelet[2039]: E0826 17:07:26.995794    2039 kubelet.go:2349] "Container runtime network not ready" networkReady="NetworkReady=false reason:NetworkPluginNotReady message:Network plugin returns error: No CNI configuration file in /etc/cni/net.d/. Has your network provider started?"
Aug 26 17:07:31 centos7-03-08 kubelet[2039]: E0826 17:07:31.997725    2039 kubelet.go:2349] "Container runtime network not ready" networkReady="NetworkReady=false reason:NetworkPluginNotReady message:Network plugin returns error: No CNI configuration file in /etc/cni/net.d/. Has your network provider started?"
Aug 26 17:07:37 centos7-03-08 kubelet[2039]: E0826 17:07:37.001876    2039 kubelet.go:2349] "Container runtime network not ready" networkReady="NetworkReady=false reason:NetworkPluginNotReady message:Network plugin returns error: No CNI configuration file in /etc/cni/net.d/. Has your network provider started?"
Aug 26 17:07:39 centos7-03-08 kubelet[2039]: I0826 17:07:39.610837    2039 kubelet.go:2110] "SyncLoop (PLEG): event for pod" pod="calico-system/calico-node-jmlbp" event=&{ID:3c950a6b-b5b3-4750-904b-ba51034aa65b Type:ContainerStarted Data:2a8b5ba6ec4332b4891c9b7a9c8ea46b9e8e42f0d464ad64fd6c2e38c837abfe}
Aug 26 17:07:42 centos7-03-08 kubelet[2039]: E0826 17:07:42.010693    2039 kubelet.go:2349] "Container runtime network not ready" networkReady="NetworkReady=false reason:NetworkPluginNotReady message:Network plugin returns error: No CNI configuration file in /etc/cni/net.d/. Has your network provider started?"
Aug 26 17:07:42 centos7-03-08 kubelet[2039]: I0826 17:07:42.636912    2039 generic.go:296] "Generic (PLEG): container finished" podID=3c950a6b-b5b3-4750-904b-ba51034aa65b containerID="2a8b5ba6ec4332b4891c9b7a9c8ea46b9e8e42f0d464ad64fd6c2e38c837abfe" exitCode=0
Aug 26 17:07:42 centos7-03-08 kubelet[2039]: I0826 17:07:42.636976    2039 kubelet.go:2110] "SyncLoop (PLEG): event for pod" pod="calico-system/calico-node-jmlbp" event=&{ID:3c950a6b-b5b3-4750-904b-ba51034aa65b Type:ContainerDied Data:2a8b5ba6ec4332b4891c9b7a9c8ea46b9e8e42f0d464ad64fd6c2e38c837abfe}
Aug 26 17:07:47 centos7-03-08 kubelet[2039]: E0826 17:07:47.011938    2039 kubelet.go:2349] "Container runtime network not ready" networkReady="NetworkReady=false reason:NetworkPluginNotReady message:Network plugin returns error: No CNI configuration file in /etc/cni/net.d/. Has your network provider started?"
Aug 26 17:07:52 centos7-03-08 kubelet[2039]: E0826 17:07:52.012964    2039 kubelet.go:2349] "Container runtime network not ready" networkReady="NetworkReady=false reason:NetworkPluginNotReady message:Network plugin returns error: No CNI configuration file in /etc/cni/net.d/. Has your network provider started?"
Aug 26 17:07:57 centos7-03-08 kubelet[2039]: E0826 17:07:57.014610    2039 kubelet.go:2349] "Container runtime network not ready" networkReady="NetworkReady=false reason:NetworkPluginNotReady message:Network plugin returns error: No CNI configuration file in /etc/cni/net.d/. Has your network provider started?"
Aug 26 17:08:02 centos7-03-08 kubelet[2039]: E0826 17:08:02.015891    2039 kubelet.go:2349] "Container runtime network not ready" networkReady="NetworkReady=false reason:NetworkPluginNotReady message:Network plugin returns error: No CNI configuration file in /etc/cni/net.d/. Has your network provider started?"
Aug 26 17:08:07 centos7-03-08 kubelet[2039]: E0826 17:08:07.016963    2039 kubelet.go:2349] "Container runtime network not ready" networkReady="NetworkReady=false reason:NetworkPluginNotReady message:Network plugin returns error: No CNI configuration file in /etc/cni/net.d/. Has your network provider started?"
Aug 26 17:08:12 centos7-03-08 kubelet[2039]: E0826 17:08:12.018804    2039 kubelet.go:2349] "Container runtime network not ready" networkReady="NetworkReady=false reason:NetworkPluginNotReady message:Network plugin returns error: No CNI configuration file in /etc/cni/net.d/. Has your network provider started?"
Aug 26 17:08:17 centos7-03-08 kubelet[2039]: E0826 17:08:17.019839    2039 kubelet.go:2349] "Container runtime network not ready" networkReady="NetworkReady=false reason:NetworkPluginNotReady message:Network plugin returns error: No CNI configuration file in /etc/cni/net.d/. Has your network provider started?"
Aug 26 17:08:20 centos7-03-08 kubelet[2039]: I0826 17:08:20.742640    2039 kubelet.go:2110] "SyncLoop (PLEG): event for pod" pod="calico-system/calico-node-jmlbp" event=&{ID:3c950a6b-b5b3-4750-904b-ba51034aa65b Type:ContainerStarted Data:25abe165088c4131b2800460e4d1b81676b78a95e0b4980735e891762fcf7fc5}
Aug 26 17:08:20 centos7-03-08 kubelet[2039]: I0826 17:08:20.744280    2039 kubelet.go:2182] "SyncLoop (probe)" probe="readiness" status="" pod="calico-system/calico-node-jmlbp"
Aug 26 17:08:20 centos7-03-08 kubelet[2039]: I0826 17:08:20.884957    2039 prober.go:121] "Probe failed" probeType="Readiness" pod="calico-system/calico-node-jmlbp" podUID=3c950a6b-b5b3-4750-904b-ba51034aa65b containerName="calico-node" probeResult=failure output=<
Aug 26 17:08:20 centos7-03-08 kubelet[2039]: calico/node is not ready: BIRD is not ready: Error querying BIRD: unable to connect to BIRDv4 socket: dial unix /var/run/bird/bird.ctl: connect: no such file or directory
Aug 26 17:08:20 centos7-03-08 kubelet[2039]: >
Aug 26 17:08:21 centos7-03-08 kubelet[2039]: I0826 17:08:21.865161    2039 prober.go:121] "Probe failed" probeType="Readiness" pod="calico-system/calico-node-jmlbp" podUID=3c950a6b-b5b3-4750-904b-ba51034aa65b containerName="calico-node" probeResult=failure output=<
Aug 26 17:08:21 centos7-03-08 kubelet[2039]: calico/node is not ready: BIRD is not ready: Error querying BIRD: unable to connect to BIRDv4 socket: dial unix /var/run/bird/bird.ctl: connect: no such file or directory
Aug 26 17:08:21 centos7-03-08 kubelet[2039]: >
Aug 26 17:08:22 centos7-03-08 kubelet[2039]: E0826 17:08:22.021854    2039 kubelet.go:2349] "Container runtime network not ready" networkReady="NetworkReady=false reason:NetworkPluginNotReady message:Network plugin returns error: No CNI configuration file in /etc/cni/net.d/. Has your network provider started?"
Aug 26 17:08:22 centos7-03-08 kubelet[2039]: I0826 17:08:22.363469    2039 prober.go:121] "Probe failed" probeType="Liveness" pod="calico-system/calico-node-jmlbp" podUID=3c950a6b-b5b3-4750-904b-ba51034aa65b containerName="calico-node" probeResult=failure output="Get \"http://localhost:9099/liveness\": dial tcp
Aug 26 17:08:22 centos7-03-08 kubelet[2039]: I0826 17:08:22.706492    2039 prober.go:121] "Probe failed" probeType="Readiness" pod="calico-system/calico-node-jmlbp" podUID=3c950a6b-b5b3-4750-904b-ba51034aa65b containerName="calico-node" probeResult=failure output=<
Aug 26 17:08:22 centos7-03-08 kubelet[2039]: calico/node is not ready: BIRD is not ready: Error querying BIRD: unable to connect to BIRDv4 socket: dial unix /var/run/calico/bird.ctl: connect: connection refused
Aug 26 17:08:22 centos7-03-08 kubelet[2039]: >
Aug 26 17:08:27 centos7-03-08 kubelet[2039]: E0826 17:08:27.023778    2039 kubelet.go:2349] "Container runtime network not ready" networkReady="NetworkReady=false reason:NetworkPluginNotReady message:Network plugin returns error: No CNI configuration file in /etc/cni/net.d/. Has your network provider started?"
Aug 26 17:08:32 centos7-03-08 kubelet[2039]: E0826 17:08:32.026055    2039 kubelet.go:2349] "Container runtime network not ready" networkReady="NetworkReady=false reason:NetworkPluginNotReady message:Network plugin returns error: No CNI configuration file in /etc/cni/net.d/. Has your network provider started?"
Aug 26 17:08:32 centos7-03-08 kubelet[2039]: I0826 17:08:32.501342    2039 kubelet.go:2182] "SyncLoop (probe)" probe="readiness" status="ready" pod="calico-system/calico-node-jmlbp"
Aug 26 17:08:37 centos7-03-08 kubelet[2039]: E0826 17:08:37.026947    2039 kubelet.go:2349] "Container runtime network not ready" networkReady="NetworkReady=false reason:NetworkPluginNotReady message:Network plugin returns error: No CNI configuration file in /etc/cni/net.d/. Has your network provider started?"

```

## Expected Behavior

After installation of calico CNI as per the document "https://projectcalico.docs.tigera.io/getting-started/kubernetes/self-managed-onprem/onpremises" the nodes on the kubernetes cluster should go into "Ready" status.

## Steps to Reproduce (for bugs)

1. Provision a kubeadm based kubernetes cluster with at least one control plane node and one worker node.
The config file whose name I supplied to the --config argument of kubeadm init command as the following data.

```
  apiVersion: kubeadm.k8s.io/v1beta3
  kind: ClusterConfiguration
  networking:
    dnsDomain: cluster.local
    serviceSubnet: 172.21.0.0/16
    podSubnet: 172.22.0.0/16

```

2. Join the worker nodes to the kubeadm based kubernetes cluster using the kubeadm join command.
3. Issue the command "kubectl create -f https://raw.githubusercontent.com/projectcalico/calico/v3.24.0/manifests/tigera-operator.yaml"
4. Issue the command "kubectl create -f /tmp/custom-resources.yaml" where the contents of the file are as follows.
```
---

  # This section includes base Calico installation configuration.
  # For more information, see: https://projectcalico.docs.tigera.io/master/reference/installation/api#operator.tigera.io/v1.Installation
  apiVersion: operator.tigera.io/v1
  kind: Installation
  metadata:
    name: default
  spec:
    # Configures Calico networking.
    calicoNetwork:
      # Note: The ipPools section cannot be modified post-install.
      ipPools:
        -
          blockSize: 26
          cidr: 172.22.0.0/16
          encapsulation: VXLANCrossSubnet
          natOutgoing: Enabled
          nodeSelector: all()
      nodeAddressAutodetectionV4:
        #interface: eth1
        cidrs:
          -
            "192.168.12.0/24"
  
---
  
  # This section configures the Calico API server.
  # For more information, see: https://projectcalico.docs.tigera.io/master/reference/installation/api#operator.tigera.io/v1.APIServer
  apiVersion: operator.tigera.io/v1
  kind: APIServer 
  metadata: 
    name: default 
  spec: {}
  
```

5.

## Your Environment

* Calico version 3.24.0
* Kubernetes version 1.24.2
* CentOS Linux release 7.9.2009 (Core)

**Comment by caseydavenport** [comment]
It sounds like BIRD might not be starting properly. To diagnose I think we'll need the logs from one of the NotReady calico/node pod instances.

**Comment by coutinhop** [comment]
This could be similar to https://github.com/projectcalico/calico/issues/6901, but we need more information to confirm that. @allankamau could you take a look and see if your setup matches? (i.e. do you have VXLAN enabled and is calico incorrectly autodetecting that tunnel's IP?)

If it is the same issue, the fixes are already in (https://github.com/projectcalico/calico/pull/6902 and https://github.com/projectcalico/calico/pull/6929), should be in a patch release soon...

**Comment by coutinhop** [comment]
Closing this as the fix is present in v3.24.4+ or v3.23.5+ (and also will be in v3.25.0+)
